### PR TITLE
Update TLS settings

### PR DESCRIPTION
**Pull request by urso**
- Add missing tls cipher suites + curve types
- Disable TLSv1.0 by default
- Update documentation

Unfortunately we don't have TLS 1.3 yet.

### CODE REVIEW

File: libbeat/docs/shared-ssl-config.asciidoc [hunk 1]
```diff
@@ -145,21 +145,28 @@ suites are used (recommended).
...
-* ECDHE-ECDSA-AES-256-GCM-SHA384 (TLS 1.2 only)
+* ECDHE-RSA-CHACHA20-POLY1205 (TLS 1.2 only)
+* ECDHE-RSA-RC4-128-SHA (disabled by default- RC4 not recommended)
+* RSA-3DES-CBC3-SHA
+* RSA-AES-128-CBC-SHA
+* RSA-AES-128-CBC-SHA256 (TLS 1.2 only, disabled by default)
+* RSA-AES-128-GCM-SHA256 (TLS 1.2 only)
+* RSA-AES-256-CBC-SHA
+* RSA-AES-256-GCM-SHA384 (TLS 1.2 only)
+* RSA-RC4-128-SHA (disabled by default - RC4 not recommended)
```
Comment (urso) [inline libbeat/docs/shared-ssl-config.asciidoc]: I sorted the list, so to not loose all sanity when double checking suite names + properties.

Comment (ph) [reply libbeat/docs/shared-ssl-config.asciidoc]: Good thing!

**Comment by ph** [comment]
Looking at this comment, TLS1.3 will be present in https://github.com/containous/traefik/issues/4275#issuecomment-453238420 and looking at the Golang tracking 1.12 is due in January 30. So could upgrade to go 1.12 in a minor (if want to) for 7.1.

File: libbeat/common/transport/tlscommon/tls_test.go [hunk 2]
```diff
@@ -160,7 +160,7 @@ func TestApplyWithConfig(t *testing.T) {
 	assert.NotNil(t, cfg.RootCAs)
 	assert.Equal(t, true, cfg.InsecureSkipVerify)
 	assert.Len(t, cfg.CipherSuites, 2)
-	assert.Equal(t, int(tls.VersionTLS10), int(cfg.MinVersion))
+	assert.Equal(t, int(tls.VersionTLS11), int(cfg.MinVersion))
 	assert.Equal(t, int(tls.VersionTLS12), int(cfg.MaxVersion))
```
Comment (ph) [inline libbeat/common/transport/tlscommon/tls_test.go]: ++ to make 1.11 default.

**Comment by urso** [comment]
> So could upgrade to go 1.12 in a minor (if want to) for 7.1.

That's the plan. Follow up issue: https://github.com/elastic/beats/issues/10226

**Pull request merged**